README.md

```diff
@@ -4,109 +4,317 @@
+Note that this command is very similar to the one above, but we are running it
+in the root of the repo, and we have removed the `--force-checkout` option. In
+this case, instead of fetching the sourced from the GitHub repository, we use
+the local source code. Also, we have used the latest commit on the local repo
+as the commit hash. However, this is not very accurate, since your local
+repository may contain uncommitted changes. This is not an issue if you are
+using the tool only for testing.
+
+## Create a custom Docker image
 
-To build the `HelloTransparentRelease` binary we use the [`cmd/builder` tool](https://github.com/project-oak/transparent-release#building-binaries-using-the-cmdbuilder-tool) from [project-oak/transparent-release](https://github.com/project-oak/transparent-release). 
+So far, we have been relying on a pre-built Maven image for building the
+binary. However, you could use a custom image with the container-based SLSA3
+builder. One way to create a custom image is to use a Dockerfile. In this
+section, we walk you through creating a custom builder image using a
+Dockerfile.
 
-### Configuring the `cmd/builder` tool
+This time, we are going to build the binary using Bazel instead of Maven.
 
-The buildconfig in [buildconfigs/hello_transparent_release.toml](buildconfigs/hello_transparent_release.toml) holds the configuration for `cmd/builder`.
+### [Optional] Build the binary from the command line
 
-### Build the `cmd/builder` tool from sources
+First, we need to make sure to have
+[Bazel set-up](https://docs.bazel.build/versions/main/tutorial/java.html#before-you-begin).
 
-Check out the [transparent-release](https://github.com/project-oak/transparent-release) repo and call:
+Then, we can build with
 
 ```bash
-transparent-release$ go run cmd/builder/main.go -build_config_path <absolute-or-relative-path-to-hello-transparent-release-repo>/hello-transparent-release/buildconfigs/hello_transparent_release.toml
+./scripts/build
 ```
 
-You will see:
+We run the binary:
 
 ```bash
-2022/09/27 17:55:29 The hash of the binary is: e8e05d1d09af8952919bf6ab38e0cc5a6414ee2b5e21f4765b12421c5db0037e
-2022/09/27 17:55:30 Storing the provenance in [..]/transparent-release/provenance.json
+./out/HelloTransparentRelease
 ```
 
-### Use the released `cmd/builder` tool
+This is the binary we want to release: our `HelloTransparentRelease` binary.
+
+Now, let's compute a sha256 digest of the `HelloTransparentRelease` binary:
 
 ```bash
-wget https://github.com/project-oak/transparent-release/releases/download/v0.1/transparent-release_0.1_Linux_x86_64.tar.gz 
-tar -xvzf transparent-release_0.1_Linux_x86_64.tar.gz
-./transparent-release  -build_config_path ./buildconfigs/hello_transparent_release.toml 
+sha256sum ./out/HelloTransparentRelease
+8a87337c16d1386510f9d3dd36a744d267945370e40c18113c78bb67e2934cae HelloTransparentRelease
 ```
 
-### Some useful flags
+This sha256 digest might or might not be the same on your machine.
```

Comment:
idea: Can we give an indication why it might not be the same? Then we can use this to highlight, what we do differntly.